### ISSUE TITLE
Disable SSL certificate validation

### ISSUE THREAD

**Issue by borlafdev** [jaimeiniesta/metainspector#128]
Hello i'm having some problems using this gem with some https url's, Can i disable SSL certificate validation?

Ruby version = ruby-2.0.0-p598
Gem & dependencies version:
  metainspector (4.4.1)
      addressable (~> 2.3.5)
      faraday (~> 0.9.0)
      faraday-cookie_jar (~> 0.0.6)
      faraday_middleware (~> 0.9.1)
      fastimage
      nokogiri (~> 1.6)

Trace: 

2015-04-27T10:46:22.077Z 19078 TID-r5kj4 WARN: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed
2015-04-27T10:46:22.078Z 19078 TID-r5kj4 WARN: /home/deploy/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/2.0.0/net/http.rb:918:in `connect'
/home/deploy/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/2.0.0/net/http.rb:918:in `block in connect'
/home/deploy/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/2.0.0/timeout.rb:52:in `timeout'
/home/deploy/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/2.0.0/net/http.rb:918:in `connect'
/home/deploy/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/2.0.0/net/http.rb:862:in `do_start'
/home/deploy/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/2.0.0/net/http.rb:851:in `start'
/home/deploy/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/2.0.0/net/http.rb:1367:in `request'
/home/deploy/.rvm/rubies/ruby-2.0.0-p598/lib/ruby/2.0.0/net/http.rb:1126:in `get'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-0.9.1/lib/faraday/adapter/net_http.rb:80:in `perform_request'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-0.9.1/lib/faraday/adapter/net_http.rb:40:in `block in call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-0.9.1/lib/faraday/adapter/net_http.rb:87:in `with_net_http_connection'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-0.9.1/lib/faraday/adapter/net_http.rb:32:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-cookie_jar-0.0.6/lib/faraday/cookie_jar.rb:22:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday_middleware-0.9.1/lib/faraday_middleware/response/follow_redirects.rb:79:in `perform_with_redirection'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday_middleware-0.9.1/lib/faraday_middleware/response/follow_redirects.rb:65:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-0.9.1/lib/faraday/request/retry.rb:110:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-0.9.1/lib/faraday/rack_builder.rb:139:in `build_response'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-0.9.1/lib/faraday/connection.rb:377:in `run_request'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/faraday-0.9.1/lib/faraday/connection.rb:140:in `get'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/metainspector-4.4.1/lib/meta_inspector/request.rb:59:in `fetch'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/metainspector-4.4.1/lib/meta_inspector/request.rb:37:in `response'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/metainspector-4.4.1/lib/meta_inspector/request.rb:21:in `initialize'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/metainspector-4.4.1/lib/meta_inspector/document.rb:36:in `new'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/metainspector-4.4.1/lib/meta_inspector/document.rb:36:in `initialize'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/metainspector-4.4.1/lib/meta_inspector.rb:20:in `new'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/metainspector-4.4.1/lib/meta_inspector.rb:20:in `new'
/home/deploy/pricescrapper/releases/20150424091227/config/initializers/litle_spider.rb:412:in `scrap_html'
/home/deploy/pricescrapper/releases/20150424091227/config/initializers/litle_spider.rb:404:in `scrap'
/home/deploy/pricescrapper/releases/20150424091227/app/jobs/scrap_link.rb:5:in `perform'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/processor.rb:75:in `execute_job'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/processor.rb:52:in `block (2 levels) in process'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/chain.rb:127:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/chain.rb:127:in `block in invoke'
/home/deploy/pricescrapper/shared/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/server/active_record.rb:6:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/chain.rb:129:in `block in invoke'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/server/retry_jobs.rb:74:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/chain.rb:129:in `block in invoke'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/server/logging.rb:11:in `block in call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/logging.rb:22:in `with_context'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/server/logging.rb:7:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/chain.rb:129:in `block in invoke'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/chain.rb:132:in `call'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/middleware/chain.rb:132:in `invoke'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/processor.rb:51:in `block in process'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/processor.rb:98:in `stats'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/sidekiq-3.3.0/lib/sidekiq/processor.rb:50:in `process'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `public_send'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `dispatch'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:122:in `dispatch'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/celluloid-0.16.0/lib/celluloid/cell.rb:60:in `block in invoke'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/celluloid-0.16.0/lib/celluloid/cell.rb:71:in `block in task'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/celluloid-0.16.0/lib/celluloid/actor.rb:357:in `block in task'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/celluloid-0.16.0/lib/celluloid/tasks.rb:57:in `block in initialize'
/home/deploy/pricescrapper/releases/20150424091227/vendor/bundle/ruby/2.0.0/gems/celluloid-0.16.0/lib/celluloid/tasks/task_fiber.rb:15:in `block in create'

**Comment by borlafdev** [comment]
I've disabled the ssl certification validation of  Farday succesfully in https://github.com/borlafdev/metainspector/blob/master/lib/meta_inspector/request.rb#L47 , i'm not familiar with the way gems handle user parameters, if I knew would make a pull request...

Is there any standard for gems?

**Comment by jaimeiniesta** [comment]
I think that we can add a new parameter to pass options to Faraday. Something like:

```ruby
MetaInspector.new('http://example.com', faraday_options: { ssl: { verify: false } }) 
```

This parameter would just be passed as received to Faraday.

I can work on this; can you pass me a sample URL to be able to reproduce this error?

Thanks!

**Comment by borlafdev** [comment]
If you could include an option like that would be very useful.

Here is the url that causes the exeption:
https://www.gasnaturaldistribucion.com/es/quiero+gas+natural/quiero+gas+para+mi+hogar/1297131284748/solicita+tu+oferta+personalizada.html

**Comment by jaimeiniesta** [comment]
I just released 4.6.0 that implements this:

https://github.com/jaimeiniesta/metainspector#disabling-ssl-verification-or-any-other-faraday-options

Thanks!
